framework/PostProcessors/Metric.py

```diff
@@ -211,6 +238,12 @@ def _handleInput(self, paramInput):
       elif child.getName() == 'Targets':
         self.targets = list(var.strip() for var in child.value.split(','))
         self.TargetsType = child.parameterValues['type']
+      elif child.getName() == 'multiOutput':
+        self.multiOutput = child.value.strip()
+      elif child.getName() == 'weight':
+        self.weight = np.asarray(list(float(var) for var in child.value.split(',')))
```

Comment:
this should be handled at the InputChecker stage... you can use the ListType.

Reply:
updated.